### PR TITLE
restore: add error field to `DownloadResponse`

### PR DESCRIPTION
**Pull request by 5kbpers**
Signed-off-by: 5kbpers <tangminghua@pingcap.com>

### What problem does this PR solve? 
For now we handle the error of downloading sst according to the error message.

### What is changed and how it works?
Handle the error of downloading sst with the `error` field of `DownloadResponse`

### Check List 

Tests 

 - Integration test

### CODE REVIEW

**Comment by kennytm** [comment]
(I've filed tikv/grpc-rs#443 for the proper solution in the future w.r.t. https://github.com/pingcap/br/pull/152#discussion_r375648138)

File: pkg/restore/backoff.go [hunk 1]
```diff
@@ -67,7 +60,7 @@ func newDownloadSSTBackoffer() utils.Backoffer {
 
 func (bo *importerBackoffer) NextBackoff(err error) time.Duration {
 	switch errors.Cause(err) {
-	case errResp, errGrpc, errEpochNotMatch:
+	case errGrpc, errEpochNotMatch, errIngestFailed:
```
Comment (overvenus) [inline pkg/restore/backoff.go]: Should we retry on`errRewriteRuleNotFound`? 

TiKV can return region not found after https://github.com/tikv/tikv/pull/7135

Comment (5kbpers) [reply pkg/restore/backoff.go]: `errRewriteRuleNotFound` is for handling with the region which exists in the range [t{tableid}, t{tableid+1}), but don't match to any rewrite rule, such as the region [t{tableid}, t{tableid}_i1). We should skip them.

Comment (overvenus) [reply pkg/restore/backoff.go]: Oops, I mean `errRegionNotFound` (`errRewriteRuleNotFound` is a typo 😂

Comment (5kbpers) [reply pkg/restore/backoff.go]: Now `errRegionNotFound` is included in `errIngestFailed`

File: go.mod [hunk 2]
```diff
@@ -42,3 +42,5 @@ require (
 	google.golang.org/api v0.14.0
 	google.golang.org/grpc v1.25.1
 )
+
+replace github.com/pingcap/kvproto v0.0.0-20200228095611-2cf9a243b8d5 => github.com/5kbpers/kvproto v0.0.0-20200316081444-1e3bcbdb1075
```
Comment (3pointer) [inline go.mod]: should we keep this change?

**Comment by kennytm** [comment]
/run-integration-test tikv=release-4.0 tidb=release-4.0 pd=release-4.0

**Comment by kennytm** [comment]
/run-integration-test 

```
[2020-03-21T11:27:24.690Z] /home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/tests/br_tls/../_utils/run_services: line 111: 14868 Aborted                 (core dumped) bin/tikv-server --pd "$PD_ADDR" -A "$TIKV_ADDR$i" --log-file "$TEST_DIR/tikv${i}.log" -C "$TIKV_CONFIG" -s "$TEST_DIR/tikv${i}"

[2020-03-21T11:29:46.107Z] Failed to start TiDB
```

**Comment by kennytm** [comment]
/run-integration-test

```
[2020-03-22T20:18:39.938Z] /home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/tests/br_tls/../_utils/run_services: line 111: 14955 Segmentation fault      (core dumped) bin/tikv-server --pd "$PD_ADDR" -A "$TIKV_ADDR$i" --log-file "$TEST_DIR/tikv${i}.log" -C "$TIKV_CONFIG" -s "$TEST_DIR/tikv${i}"

[2020-03-22T20:18:49.892Z] /home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/tests/br_tls/../_utils/run_services: line 111: 14957 Aborted                 (core dumped) bin/tikv-server --pd "$PD_ADDR" -A "$TIKV_ADDR$i" --log-file "$TEST_DIR/tikv${i}.log" -C "$TIKV_CONFIG" -s "$TEST_DIR/tikv${i}"

[2020-03-22T20:21:11.330Z] Failed to start TiDB
```

**Comment by overvenus** [comment]
/run-integration-test pd=release-4.0 tikv=release-4.0 tidb=release-4.0

**Pull request merged**